src/web/crate_details.rs

```diff
@@ -1035,13 +1053,7 @@ mod tests {
             assert!(body.contains("<a href=\"/crate/dummy/latest/source/\""));
             assert!(body.contains("<a href=\"/crate/dummy/latest\""));
 
-            assert_redirect_cached(
```

Comment:
iron & axum both handle this by returning a permanent redirect. 

Difference is that iron still applied middleware after this, and axum does not. 

Difference is that axum does not apply middleware to this specific redirect. 

I'm trying to find out why, but I also believe for this specific case it doesn't matter: 

It's a permanent redirect, and I imagine that weighs higher than the cache-control header.

Reply:
Just asked in the axum channel, this is fixed in axum 0.6 (which is rc6 right now, near to releasing)

Reply:
Looks like axum 0.6 is released now: https://docs.rs/axum/0.6.0/axum/index.html